### ISSUE TITLE
Is there any problem with these set of syntaxes(strictness problem) here?

### ISSUE THREAD

**Issue by liyili2** [kframework/k-legacy#590]
I test this set of syntax declarations:

```
     syntax KItem ::= reval(K) [semantic,klabel('reval)]
     syntax KItem ::= peval(K) [semantic]

     //TODO: might not need this list in the future. 
     syntax HeatItem ::= hItem(K) [strict(1(context('reval)))]
     syntax HeatList ::= List{HeatItem,"hs::"} [strict]
     syntax KItem ::= hlist(HeatList) [strict]
     syntax HeatList ::= toHeatList(List) [function]

```

It gives me this output in java backend.

```
Exception in thread "main" java.lang.NullPointerException
    at org.kframework.compile.transformers.StrictnessToContexts.visit(StrictnessToContexts.java:275)
    at org.kframework.compile.transformers.StrictnessToContexts.visit(StrictnessToContexts.java:24)
    at org.kframework.kil.Module.accept(Module.java:96)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitList(AbstractVisitor.java:135)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:284)
    at org.kframework.kil.Definition.accept(Definition.java:185)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.compile.utils.CompilerTransformerStep.compile(CompilerTransformerStep.java:24)
    at org.kframework.compile.utils.CompilerSteps.compile(CompilerSteps.java:44)
    at org.kframework.kompile.KompileFrontEnd.genericCompile(KompileFrontEnd.java:192)
    at org.kframework.kompile.KompileFrontEnd.kompile(KompileFrontEnd.java:161)
    at org.kframework.kompile.KompileFrontEnd.main(KompileFrontEnd.java:72)
    at org.kframework.main.Main.main(Main.java:49)
```

**Comment by liyili2** [comment]
I mean I got the idea why this set of syntaxes is wrong, but I think we should add enough exception capturing here.

**Comment by yilongli** [comment]
This is not a bug specific to the Java backend. The pass `StrictnessToContexts` is shared between different backends.

**Comment by dwightguth** [comment]
This should be fixed by #634. Moving to other milestone and de-assigning for now.
